static/js/components/hang-punctuation.js

```diff
@@ -0,0 +1,57 @@
+class HangPunctuation {
+  constructor() {
+    this.containers = Array.from(document.querySelectorAll('.js-hang-punc'));
+    this.punctuationMarks = {
+      '\u201c': 'medium', // “ - ldquo - left smart double quote
+      '\u2018': 'small', // ‘ - lsquo - left smart single quote
+      '\u0022': 'medium', // " - ldquo - left dumb double quote
+      '\u0027': 'small', // ' - lsquo - left dumb single quote
+      '\u00AB': 'large', // « - laquo - left double angle quote
+      '\u2039': 'medium', // ‹ - lsaquo - left single angle quote
+      '\u201E': 'medium', // „ - bdquo - left smart double low quote
+      '\u201A': 'small', // ‚ - sbquo - left smart single low quote
+    };
+
+    if (this.containers.length > 0) {
+      this.containers.forEach(this.hangPunc.bind(this));
+    }
+  }
+
+  hangPunc(container) {
+    const containerChildren = container.childNodes;
+
+    // Loop over all direct descendants of the $container
+    // If it's a blockquote, loop over its direct descendants
+    for (let i = 0; i < containerChildren.length; i += 1) {
+      const el = containerChildren[i];
+
+      for (let k = 0; k < el.childNodes.length; k += 1) {
```

Comment:
This is the only change to this file. The original contains an if/else about not using childnodes if it is a blockquote, but with Craft they are ALWAYS childnodes.